FailedUnrecoverable sectors not remove

lotus-miner sectors remove --really-do-it 0

ERROR: normal shutdown of state machine

Daemon: 1.2.1+git.375b7a1c4+api1.0.0
Local: lotus version 1.2.1+git.375b7a1c4

please tell me how to remove FailedUnrecoverable sectors, thanks

lotus-miner sectors update-state --really-do-it 373 Removed
ERROR: normal shutdown of state machine

2020-11-26T19:07:17.134+0800    WARN    rpc     go-jsonrpc@v0.1.2-0.20201116213214-f4b2d34247f0/handler.go:273  error in RPC call to 'Filecoin.SectorsUpdate': normal shutdown of state machine:
    github.com/filecoin-project/go-statemachine.init
        /root/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/machine.go:16

--- a/cmd/lotus-storage-miner/info.go
+++ b/cmd/lotus-storage-miner/info.go
@@ -253,6 +253,7 @@ var stateList = []stateMeta{
 
 	{col: color.FgRed, state: sealing.UndefinedSectorState},
 	{col: color.FgYellow, state: sealing.Packing},
+	{col: color.FgYellow, state: sealing.GetTicket},
 	{col: color.FgYellow, state: sealing.PreCommit1},
 	{col: color.FgYellow, state: sealing.PreCommit2},
 	{col: color.FgYellow, state: sealing.PreCommitting},

--- a/extern/storage-sealing/fsm.go
+++ b/extern/storage-sealing/fsm.go
@@ -159,8 +159,12 @@ var fsmPlanners = map[SectorState]func(events []statemachine.Event, state *Secto
 		on(SectorFaultReported{}, FaultReported),
 	),
 
+	FaultReported: final, // not really supported right now
+
 	FaultedFinal: final,
 	Removed:      final,
+
+	FailedUnrecoverable: final,
 }
 
 func (m *Sealing) plan(events []statemachine.Event, state *SectorInfo) (func(statemachine.Context, SectorInfo) error, uint64, error) {

--- a/extern/storage-sealing/fsm_test.go
+++ b/extern/storage-sealing/fsm_test.go
@@ -160,3 +160,18 @@ func TestPlanCommittingHandlesSectorCommitFailed(t *testing.T) {
 
 	require.Equal(t, CommitFailed, m.state.State)
 }
+
+func TestPlannerList(t *testing.T) {
+	for state := range ExistSectorStateList {
+		_, ok := fsmPlanners[state]
+		require.True(t, ok, "state %s", state)
+	}
+
+	for state := range fsmPlanners {
+		if state == UndefinedSectorState {
+			continue
+		}
+		_, ok := ExistSectorStateList[state]
+		require.True(t, ok, "state %s", state)
+	}
+}

--- a/extern/storage-sealing/sector_state.go
+++ b/extern/storage-sealing/sector_state.go
@@ -76,7 +76,7 @@ const (
 
 func toStatState(st SectorState) statSectorState {
 	switch st {
-	case Empty, WaitDeals, Packing, GetTicket, PreCommit1, PreCommit2, PreCommitting, PreCommitWait, WaitSeed, Committing, CommitWait, FinalizeSector:
+	case Empty, WaitDeals, Packing, GetTicket, PreCommit1, PreCommit2, PreCommitting, PreCommitWait, WaitSeed, Committing, SubmitCommit, CommitWait, FinalizeSector:
 		return sstSealing
 	case Proving, Removed, Removing:
 		return sstProving

--- a/extern/storage-sealing/sector_state.go
+++ b/extern/storage-sealing/sector_state.go
@@ -6,6 +6,7 @@ var ExistSectorStateList = map[SectorState]struct{}{
 	Empty:                {},
 	WaitDeals:            {},
 	Packing:              {},
+	GetTicket:            {},
 	PreCommit1:           {},
 	PreCommit2:           {},
 	PreCommitting:        {},
@@ -75,7 +76,7 @@ const (
 
 func toStatState(st SectorState) statSectorState {
 	switch st {
-	case Empty, WaitDeals, Packing, PreCommit1, PreCommit2, PreCommitting, PreCommitWait, WaitSeed, Committing, CommitWait, FinalizeSector:
+	case Empty, WaitDeals, Packing, GetTicket, PreCommit1, PreCommit2, PreCommitting, PreCommitWait, WaitSeed, Committing, CommitWait, FinalizeSector:
 		return sstSealing
 	case Proving, Removed, Removing:
 		return sstProving

